src/asciidoctor-kroki.js

```diff
@@ -67,13 +71,16 @@ const processKroki = (processor, parent, attrs, diagramType, diagramText, contex
   delete blockAttrs.title
   delete blockAttrs.caption
   delete blockAttrs.opts
-  const inline = attrs['inline-option'] === ''
-  if (inline) {
-    blockAttrs['inline-option'] = ''
-  }
-  const interactive = attrs['interactive-option'] === ''
-  if (interactive) {
-    blockAttrs['interactive-option'] = ''
+
+  const opt = choices.reduce((accum, [name, option]) => {
+    return attrs[name + '-option'] === '' ? option : accum
+  }, undefined) ||
+      choices.reduce((accum, [name, option]) => {
+        return parent.document.getAttribute('kroki-default-options') === name ? option : accum
+      }, undefined)
```

Comment:
Not sure about this code, it might be elegant but I cannot understand it. If we want to make the code more concise I guess we could introduce a function to check a given list of option but if we can avoid using `reduce` that would be great (since it's not consistent with the code style and not straightforward to read in my opinion)

Reply:
Indeed, it might be write-only code, like xslt :-) It checks for whether any block/macro attribute option is set, and if none, it checks for the global/default option, and returns the appropriate option name.  I'll see if I can make it more transparent :-)  but it was the simplest code I could actually get to work.

Reply:
See how you like it now...